shared/cluster.go

```diff
@@ -584,3 +584,75 @@ func processNodeArgs(nodeArgs string) (nodeArgsMapSlice []map[string]string) {
 
 	return nodeArgsMapSlice
 }
+
+// DeleteClusterNode deletes a node from the cluster filtering the name out by the IP.
+func DeleteClusterNode(ip string) error {
+	if ip == "" {
+		return ReturnLogError("must send a ip: %s\n", ip)
+	}
+
+	name, err := GetNodeNameByIP(ip)
+	if err != nil {
+		return ReturnLogError("failed to get node name by ip: %w\n", err)
+	}
+
+	res, delErr := RunCommandHost("kubectl delete node " + name + " --wait=false  --kubeconfig=" + KubeConfigFile)
+	if delErr != nil {
+		return ReturnLogError("failed to delete node: %w\n", delErr)
+	}
+	LogLevel("info", "Deleting node: %s", res)
+
+	// delay not meant to wait if node is deleted
+	// but rather to give time for the node to be removed from the cluster
+	delay := time.After(20 * time.Second)
```

Comment:
we do need unfortunately i did not found any good  conditional to wait  without adding much complexity 
and this is actually just a delay to return, i really prefer delay rather than sleep

Reply:
yeah I understand the need.... I think it's fine for now but might be something we can improve upon later. Ideally no waits/sleeps in the code at all but of course because of testing some areas require it 😞